app/src/lib/shells/win32.ts

```diff
@@ -215,6 +245,193 @@ async function findGitBash(): Promise<string | null> {
   return null
 }
 
+async function processWslRegistryKeys(keys: string[]): Promise<void> {
+  keys.forEach(async function(key) {
+    const path = enumerateValues(HKEY.HKEY_CURRENT_USER, key)
+
+    if (path) {
+      const distributionName = path.find(e => e.name === 'DistributionName')
+      const basePath = path.find(e => e.name === 'BasePath')
+
+      if (
+        basePath &&
+        basePath.type === RegistryValueType.REG_SZ &&
+        distributionName &&
+        distributionName.type === RegistryValueType.REG_SZ
+      ) {
+        const path = Path.join(
+          `${basePath.data}`,
+          `${distributionName.data}${'.exe'}`
+        )
+
+        if (await pathExists(path)) {
+          WslShells.push({
+            shell: Shell.WslBash,
+            name: `WSL Bash (${distributionName.data})`,
+            path: path,
+          })
+        } else {
+          const defaultInstallPath = Path.join(
+            process.env.HOME + '\\AppData\\Local\\Microsoft\\WindowsApps\\',
+            distributionName.data + '.exe'
+          )
+          if (await pathExists(defaultInstallPath)) {
+            WslShells.push({
+              shell: Shell.WslBash,
+              name: `WSL Bash (${distributionName.data})`,
+              path: defaultInstallPath,
+            })
+          }
+        }
+      }
+    }
+  })
+}
+
+async function enumerateWslShellNames(): Promise<void> {
+  const promise = new Promise<void>(async resolve => {
+    const hkeyCurrentUser = 'HKEY_CURRENT_USER'
+    const keyPath = `${hkeyCurrentUser}\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Lxss`
+
+    // forced to used reg.exe to enumerate folder names as registry-js doesnt support folders
```

Comment:
I needed to enumerate folders in the registry to get the installed WSL Distros (commercial and custom) - as far as i could tell registry-js only supported enumerating keys under a folder and not the folders. I was wondering about updating the registry-js package to do this, but was unsure. Perhaps this would be an idea before proceeding with this PR.

Reply:
Does the work in this PR help with your case? https://github.com/desktop/registry-js/pull/33 If so, I can prioritize reviewing that and getting an release out for you to test.

Reply:
Will take a look and see - thanks!

Reply:
Yea this looks like it will provide the enum of folders (keys) in the registry. Once this is reviewed/merged i will update to the latest package and test.